### PR TITLE
Extract RespondWorkflowTaskFailedRequest handler

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Extracted handler for `RespondWorkflowTaskFailedRequest` out of `workflow_task_handler_callbacks.go` into its own file.

Remaining ones will be moved in follow-up PRs.

## Why?

There's no clear benefit to grouping the WFT API handlers all in the - poorly named - file. The PR makes this consistent with the existing API handlers now.

## How did you test it?

Pure refactoring, no behavior change.

### CODE REVIEW

File: service/history/api/respondworkflowtaskfailedrequest/api.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package respondworkflowtaskfailedrequest
```
Comment (yycptt) [inline service/history/api/respondworkflowtaskfailedrequest/api.go:25]:
```suggestion
package respondworkflowtaskfailed
```

**Review by yycptt — APPROVED** [review]
I just skimmed through the change. I assume it's mostly copy-paste. :)

**Review by dandavison — APPROVED** [review]
Nice! (See package rename thread that @yycptt started).

> I assume it's mostly copy-paste. 

Btw git's [color-moved](https://git-scm.com/docs/git-config#Documentation/git-config.txt-diffcolorMoved) feature is useful for these. E.g.

```
gh pr checkout 5902
```

```gitconfig
[diff]
    colorMoved = default
```

File: service/history/api/respondworkflowtaskfailedrequest/api.go (hunk 1 above)
Comment (dandavison) [reply service/history/api/respondworkflowtaskfailedrequest/api.go:25]: Actually, why "respond"? Should it be simply `workflowtaskfailed`?

```
temporal(main અ ) ls service/history/api
activity_util.go              get_history_util.go                 listtasks                     removesignalmutablestate      retry_util.go            update_workflow_util.go
addtasks                      get_workflow_util.go                multioperation                replication                   signal_workflow_util.go  updateworkflow
consistency_checker.go        getdlqtasks                         namespace.go                  replication_util.go           signalwithstartworkflow  verifychildworkflowcompletionrecorded
consistency_checker_test.go   getworkflowexecutionhistory         pollupdate                    replicationadmin              signalworkflow           workflow_id_dedup.go
create_workflow_util.go       getworkflowexecutionhistoryreverse  queryworkflow                 requestcancelworkflow         startworkflow            workflow_lease.go
deletedlqtasks                getworkflowexecutionrawhistory      reapplyevents                 resetstickytaskqueue          task_category.go
deleteworkflow                getworkflowexecutionrawhistoryv2    recordactivitytaskheartbeat   resetworkflow                 task_category_test.go
describemutablestate          isactivitytaskvalid                 recordactivitytaskstarted     respondactivitytaskcanceled   terminateworkflow
describeworkflow              isworkflowtaskvalid                 recordchildworkflowcompleted  respondactivitytaskcompleted  token.go
forcedeleteworkflowexecution  listqueues                          refreshworkflow               respondactivitytaskfailed     trim_history_util.go
```

Comment (stephanos) [reply service/history/api/respondworkflowtaskfailedrequest/api.go]: Yup, 100% copy-paste; including the package name ;) Fixed!

**Pull request merged**